docs: Improve Thanos Ruler doc with examples (#5723)
prometheus-operator/prometheus-operator: Documentation/thanos.md

## Thanos Ruler

> The [Thanos Ruler](https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md) component evaluates Prometheus recording and alerting rules against chosen query API. A `ThanosRuler` instance requires at least one `queryEndpoints` which points to the location of Thanos Queriers. The `queryEndpoints` are used to configure the `--query` arguments(s) of the Thanos rulers. The `alertmanagersConfig` allows you to provide the Alertmanager definition, so you can use Ruler to evaluate alerting rules defined on `PrometheusRule`. The `alertmanagersConfig` then will allow you to define your `--alertmanagers.config` argument via secret.

Reviewer: (nit) I would link to https://thanos.io/tip/components/rule.md/ instead of the GitHub Markdown file.

Author: hi @simonpasquier thanks for the feedback, applied the changes you suggested :bowing_woman:

> The [Thanos Ruler](https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md) component evaluates Prometheus recording and alerting rules against chosen query API. A `ThanosRuler` instance requires at least one `queryEndpoints` which points to the location of Thanos Queriers. The `queryEndpoints` are used to configure the `--query` arguments(s) of the Thanos rulers. The `alertmanagersConfig` allows you to provide the Alertmanager definition, so you can use Ruler to evaluate alerting rules defined on `PrometheusRule`. The `alertmanagersConfig` then will allow you to define your `--alertmanagers.config` argument via secret.

Reviewer: (suggestion)
```suggestion
The [Thanos Ruler](https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md) component evaluates Prometheus recording and alerting rules against chosen query API. A `ThanosRuler` instance requires at least one Query API server defined either by the `.spec.queryConfig` field or the `.spec.queryEndpoints` field.  It can also be configured to send alerts to Alertmanager with the `.spec.alertmanagersConfig`.
```

> ## Thanos Ruler

Reviewer 2: Hey @heliapb thanks for taking care of this.

Just to fix the current CI issues, could you run the following command?

```bash
make --always-make format generate
```

This will update the and format auto-generated docs

Author: done, pls review

The [Thanos Ruler](https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md) component allows recording and alerting rules to be processed across

> multiple Prometheus instances. A `ThanosRuler` instance requires at least one `queryEndpoints` which points to the location of Thanos Queriers or Prometheus instances. The `queryEndpoints` are used to configure the `--query` arguments(s) of the Thanos rulers. The `alertmanagersConfig` allows you to provide the Alertmanager definition, so you can use Ruler to evaluate alerting rules defined on `PrometheusRule`. The `alertmanagersConfig` then will allow you to define your `--alertmanagers.config` argument via secret.

Reviewer 2: I might be wrong, but I guess the `queryEndPoints` can't point directly to a Prometheus instance, because I think it's using the Store API which is only exposed by Thanos Components, on that case, it could point to Thanos Query, or Thanos Sidecar.

Author: Hi @nicolastakashi fixed, please review.

      role: my-thanos-rules
  queryEndpoints:
    - dnssrv+_http._tcp.my-thanos-querier.monitoring.svc.cluster.local
  alertmanagersConfig:
    key: alertmanager-configs.yaml
    name: thanosruler-alertmanager-config
```

More context for your Alertmanager configuration can be found in the [Thanos documentation](https://thanos.io/tip/components/rule.md/#alertmanager). An example:

```yaml
 alertmanagers:

> - static_configs: ["alertmanager-dns"]

Reviewer: to make it consistent with the query endpoints
```suggestion
- static_configs:
  - "dnssrv+_web._tcp.alertmanager-operated.monitoring.svc.cluster.local"
```

>   api_version: v1

Reviewer: (suggestion) everybody should use v2 by now :)
```suggestion
  api_version: v2
```

```

Can be saved as `/tmp/alertmanager-configs.yaml`, and you can create in your namespace, for example `monitoring` as `thanosruler-alertmanager-config` imperatively with:

```sh

> kubectl -n monitoring create secret generic thanosruler-alertmanager-config --from-file=alertmanager-configs.yaml=/tmp/alertmanager-configs.yaml

Reviewer: we can also use stringData to show the final secret in clear-text:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: thanosruler-alertmanager-config
stringData:
  alertmanager-configs.yaml: |-
    alertmanagers: ...
```
